### PR TITLE
chore: make the territory infobar pattern more strict, add tests [skip ci]

### PR DESCRIPTION
**Pull request by cmkohnen**
Stumbled upon this while trying to implement something else, feel free to rename title, as I'm not very familiear with conventional commit messages in this situation.

### CODE REVIEW

File: fabric/src/test/java/TestRegex.java [hunk 1]
```diff
@@ -350,6 +351,14 @@ public void GuildRankReplacementFeature_RECRUIT_USERNAME_PATTERN() {
         p.shouldMatch("§3[_user0name_");
     }
 
+    @Test
+    public void InfoBar_TERRITORY_INFO_PATTERN() {
+        PatternTester p = new PatternTester(InfoBar.class, "TERRITORY_INFO_PATTERN");
+        p.shouldMatch("§aLutho§2 [PROF]");
+        p.shouldMatch("§bCorkus City§3 [HOC]");
+        p.shouldMatch("§cDetlas§4 [AVO]");
```
Comment (magicus) [inline fabric/src/test/java/TestRegex.java:371]: It seems form your changes to the regex that you got a match when you did not expect one. If so, please add a negative test case here that shows that your changes fix it.

File: common/src/main/java/com/wynntils/models/worlds/bossbars/InfoBar.java [hunk 2]
```diff
@@ -13,13 +13,13 @@
...
-    // §cDetlas§4 [AVO]
+    // Test in InfoBar_TERRITORY_INFO_PATTERN
     private static final Pattern TERRITORY_INFO_PATTERN =
-            Pattern.compile("§.(?<territory>.+)§. \\[(?<tag>[A-Za-z]{3,4})\\]");
+            Pattern.compile("§[abc](?<territory>.+)§[234] \\[(?<tag>[A-Za-z]{3,4})\\]");
 
     // §7Lv. 92§f - §bKingdom Foxes§f - §762% XP
     private static final Pattern GUILD_INFO_PATTERN =
-            Pattern.compile("§7Lv. (?<level>\\d+)§f - §b(?<guild>.+)§f - §7(?<xp>\\d+)% XP");
+            Pattern.compile("§7Lv\\. (?<level>\\d+)§f - §b(?<guild>[a-zA-Z ]+)§f - §7(?<xp>\\d+)% XP");
```
Comment (kristofbolyai) [inline common/src/main/java/com/wynntils/models/worlds/bossbars/InfoBar.java]:
```suggestion
            Pattern.compile("§7Lv\\. (?<level>\\d+)§f - §b(?<guild>[a-zA-Z\\s]+)§f - §7(?<xp>\\d+)% XP");
```
Perhaps this is a bit cleaner.

**Comment by cmkohnen** [comment]
Oh sorry for the confusion. I noticed that the . in Lv. wasn't properly escaped and a wildcard was used for the color codes. I hence added this test for the three different cases (your guild, other guild, allied guild).

Am 21. Mai 2024 13:11:32 MESZ schrieb Magnus Ihse Bursie ***@***.***>:
>@magicus commented on this pull request.
>
>
>
>> @@ -350,6 +351,14 @@ public void GuildRankReplacementFeature_RECRUIT_USERNAME_PATTERN() {
>         p.shouldMatch("§3[_user0name_");
>     }
> 
>+    @Test
>+    public void InfoBar_TERRITORY_INFO_PATTERN() {
>+        PatternTester p = new PatternTester(InfoBar.class, "TERRITORY_INFO_PATTERN");
>+        p.shouldMatch("§aLutho§2 [PROF]");
>+        p.shouldMatch("§bCorkus City§3 [HOC]");
>+        p.shouldMatch("§cDetlas§4 [AVO]");
>
>It seems form your changes to the regex that you got a match when you did not expect one. If so, please add a negative test case here that shows that your changes fix it.
>
>-- 
>Reply to this email directly or view it on GitHub:
>https://github.com/Wynntils/Artemis/pull/2514#pullrequestreview-2068331579
>You are receiving this because you authored the thread.
>
>Message ID: ***@***.***>
-- Von /e/OS E-Mail gesendet.

**Comment by kristofbolyai** [comment]
So this isn't a fix to a concrete issue? Okay, we'll accept this now, but this PR dances on the edge of "change just to change something". Obviously, from your previous contributions, I know this is not just so you have commits in the repository. Perhaps, small stuff like this, can be bundled in your next bigger PR, we usually do that too.

Anyways thank you for this and the tests :)

**Comment by cmkohnen** [comment]
> So this isn't a fix to a concrete issue? Okay, we'll accept this now, but this PR dances on the edge of "change just to change something". Obviously, from your previous contributions, I know this is not just so you have commits in the repository. Perhaps, small stuff like this, can be bundled in your next bigger PR, we usually do that too.

Sorry, I maintain a project, in which commit history isn't that important to us and am not familiar with Artemis' handling of such stuff, I can implement the feature I initially planned in this branch if you want, so it doesn't have to warrant a commit for such a small change.

**Comment by kristofbolyai** [comment]
If you fix my review, we can merge this. It's nothing to be sorry about, just noting it, in case you contribute in the future.

**Comment by cmkohnen** [comment]
Notice that I also updated the patterns in GuildModel.java, which also use a space to check for Guild Names and convert the comments to actual tests.

**Comment by magicus** [comment]
(Actually, I prefer to have several smaller PRs than bundling different fixes in a single PR.)

**Comment by kristofbolyai** [comment]
> (Actually, I prefer to have several smaller PRs than bundling different fixes in a single PR.)

Yes, but I also want don't want to have so many "fix typo" commits floating around.

**Review by kristofbolyai — APPROVED** [review]

**Comment by kristofbolyai** [comment]
Thank you, looks good.

**Review by magicus — APPROVED** [review]
Thanks for improving the regexes!

**Pull request merged**